samza-core/src/main/java/org/apache/samza/storage/TaskSideInputStorageManager.java

```diff
@@ -258,9 +142,10 @@ private void initializeStoreDirectories() {
   /**
    * Writes the offset files for all side input stores one by one. There is one offset file per store.
    * Its contents are a JSON encoded mapping from each side input SSP to its last processed offset, and a checksum.
+   *
+   * @param lastProcessedOffsets The offset per SSP to write
    */
-  @VisibleForTesting
-  void writeOffsetFiles() {
+  public void writeOffsetFiles(Map<SystemStreamPartition, String> lastProcessedOffsets) {
```

Comment:
nit: now that you are changing the signature anyways, can we rename this to `writeFileOffsets` so that is consistent with `getFileOffsets`.

what do you think?